styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -365,17 +365,24 @@ private synchronized void tryPollPods() {
     final int resourceVersion = Integer.parseInt(list.getMetadata().getResourceVersion());
 
     for (Pod pod : list.getItems()) {
+      // Emit events
       logEvent(Watcher.Action.MODIFIED, pod, resourceVersion, true);
-      inspectPod(Watcher.Action.MODIFIED, pod);
+      final boolean isUnwantedPod = inspectPod(Watcher.Action.MODIFIED, pod);
+      final boolean isStyxPod = pod.getMetadata().getName().startsWith(STYX_RUN);
+
+      if (isUnwantedPod && isStyxPod) {
+        LOG.info("Deleting unwanted pod {}", pod.getMetadata().getName());
+        client.pods().delete(pod);
+      }
     }
   }
 
-  private void inspectPod(Watcher.Action action, Pod pod) {
+  private boolean inspectPod(Watcher.Action action, Pod pod) {
```

Comment:
Maybe splitting it into two methods? The first one simply does those `if` checks and the second one does translation if the first one passes.

Something like:

```
if (isUnwantedStyxPod(...)) {
        LOG.info("Deleting unwanted pod {}", pod.getMetadata().getName());
        client.pods().delete(pod);
} else {
        doTranslation(...);
}
```